lib/remote/apilistener-configsync.cpp

```diff
@@ -462,3 +477,37 @@ void ApiListener::SendRuntimeConfigObjects(const JsonRpcConnection::Ptr& aclient
 	Log(LogInformation, "ApiListener")
 		<< "Finished syncing runtime objects to endpoint '" << endpoint->GetName() << "'.";
 }
+
+/**
+ * Wait for the specified object name of the given type to become available for processing.
+ *
+ * @param Type::Ptr ptype The type of the object you want to wait for
+ * @param String objName The object name you want to wait for
+ */
+void ObjectNameMutex::Lock(const Type::Ptr& ptype, const String& objName)
+{
+	std::unique_lock<std::mutex> lock(m_Mutex);
+	m_CV.wait(lock, [this, &ptype, &objName]{
+		auto& locked = m_LockedObjectNames[ptype.get()];
+		return locked.find(objName) != locked.end();
+	});
+
+	// Add object name to the locked list again to block all other threads that try
+	// to process a message affecting the same object.
+	m_LockedObjectNames[ptype.get()].emplace(objName);
+}
+
+/**
+ * Remove the specified object name of the given type from the locked list and notify all waiting threads.
+ *
+ * @param Type::Ptr ptype The type of the object you want to unblock
+ * @param String objName The name of the object you want to unblock
+ */
+void ObjectNameMutex::UnLock(const Type::Ptr& ptype, const String& objName)
```

Comment:
Please call it `Unlock()` as that's a single word and is also how the method is capitalized in `ObjectLock`.